### ISSUE TITLE
[BUG][R] query parameters are not considered while making the request 

### ISSUE THREAD

**Issue by Ramanth** [OpenAPITools/openapi-generator#2212]
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] What's the version of OpenAPI Generator used?
- [x] Have you search for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Bounty to sponsor the fix ([example](https://www.bountysource.com/issues/66123212-javascript-client-produces-a-wrong-object-for-a-string-enum-type-that-is-used-with-ref))

##### Description
The query parameters are not being considered in the api calls.
##### openapi-generator version

latest from r_refactor branch (4.0.0)

##### Command line used for generation
```
java -jar openapi-generator-cli.jar generate --input-spec http://localhost:8000/xxxxxxxxxxx/xxx/xx -g r 
--output OutFolder --config config.json 
```
##### Steps to reproduce
hit any endpoint from the R client with few query parameters. 
The query params are not being included in the request.
##### Related issues/PRs
Not yet
##### Suggest a fix
currently in the api_client.mustache file the lines for the httr request are as below
## current code
```
httr::POST(url, queryParams, headers, body = body, httr::content_type("application/json"), ...)
```
so the above should be modified to include the name of the parameter "query" for queryParams as below
## fix
```
httr::POST(url, query = queryParams, headers, body = body, httr::content_type("application/json"), ...)
```
This shoule be done for other http methods like GET, PUT, DELETE..etc.,

**Comment by wing328** [comment]
You're right. According to the doc: https://www.rdocumentation.org/packages/httr/versions/1.4.0/topics/POST
```
... Further named parameters, such as query, path, etc, passed on to modify_url(). Unnamed parameters will be combined with config().
```

**Comment by wing328** [comment]
Fixed via #2214
